shell/platform/android/io/flutter/embedding/android/FlutterSurfaceView.java

```diff
@@ -56,6 +58,35 @@ public void surfaceCreated(@NonNull SurfaceHolder holder) {
 
           if (shouldNotify()) {
             connectSurfaceToRenderer();
+
+            // Workaround for https://github.com/flutter/flutter/issues/139630
+            // We should not need this if Samsung fixes b/316626640
+            // This requires painful manually testing to reproduce.
+            // Run the default counter application on a Samsung phone running
+            // on Anroid 14. Background the application. Wait a few minutes.
+            // Resume the application (bring it to the foreground again).
+            // Sometimes, on resuming, the surface frame has been marked as
+            // not visible and renders a black screen. Rotating the device will
+            // resolve this, or requesting a layout on the surface view.
+            // However, requesting layout on the very next frame is not enough.
+            // Requesting it on the next three frames has locally made the bug
+            // fail to reproduce.
+            if (Build.VERSION.SDK_INT >= 34 && Build.MANUFACTURER.equals("samsung")) {
+              Choreographer.getInstance()
+                  .postFrameCallback(
+                      new Choreographer.FrameCallback() {
+                        int count = 3;
+
+                        @Override
+                        public void doFrame(long frameTimeNanos) {
+                          requestLayout();
+                          count--;
+                          if (count > 0) {
+                            Choreographer.getInstance().postFrameCallback(this);
+                          }
+                        }
+                      });
```

Comment:
Are we sure we want to do this? This is basically saying that Samsung devices get handled differently than other devices. This is going to lead to situations where things can reproduce in one device but not another.

Can we do something like detect the failure mode and do the hack when we detect the failure mode, instead?

Reply:
We're currently in a situation where things reproduce in one device but not another.

We cannot detect the failure mode here - the places where the failure is visible are all internal to SurfaceFlinger and not accessible from Java or C++ based app code on Android.